### PR TITLE
Provide reprs for basic types.

### PR DESCRIPTION
**Pull request by Carreau**
Prototype.

For a couple of object (types, function, methods, module), replace the
raw repr, with an HTML one, which is details+summary, where the
collapsed version is identical to the plain-text one + the small
collapsed indicator.

Need changes to IPython  to work, plese do nt merge yet.

### CODE REVIEW

**Comment by Carreau** [comment]
disp-basetype

**Review by rgbkrk — APPROVED** [review]
Loving it!

**Comment by Carreau** [comment]
Pushed a big cleanup. It's not the best but it does get the job done. I'm happy to get that out. 

For the park repr, if you want them to abort on newer version of pyspak, we can just check from `hasattr(Obj, '_repr_html_')` and `return None` if it's the case. THat should work.

File: disp/py3only.py [hunk 1]
```diff
@@ -1,29 +1,66 @@
+"""
+This submodule contains formatting utilities and formatters which will work only
+on Python 3.6+. There is no inherent reasons why it would not work on earlier
+version of Python, it just makes use of features that are 3.6 only – Like
+f-strings – to make the code more readable. Feel free to send patches that makes
+it compatible with earlier versions of python
```
Comment (rgbkrk) [inline disp/py3only.py:6]: 👍

**Comment by rgbkrk** [comment]
Do you still need changes to IPython for this or is it fit to ship?

**Comment by Carreau** [comment]
It should work on stable IPython I've vendors my changes. At least I hope.
I haven't tried.

**Comment by rgbkrk** [comment]
We should have a low number of users if we ship as is here. :wink:

**Pull request merged**